pkg/ebpf/tracee.go

```diff
@@ -179,12 +185,40 @@ type Tracee struct {
 	containers        *containers.Containers
 	procInfo          *procinfo.ProcInfo
 	eventsSorter      *sorting.EventsChronologicalSorter
+	kernelSymbols     *helpers.KernelSymbolTable
 }
 
 func (t *Tracee) Stats() *metrics.Stats {
 	return &t.stats
 }
 
+func isEssentials(id int32) bool {
+	return EventsDefinitions[id].EssentialEvent
+}
+
+func setEssential(id int32) {
+	event := EventsDefinitions[id]
+	event.EssentialEvent = true
+	EventsDefinitions[id] = event
+}
+
+func (t *Tracee) handleEventsDependencies(e int32) {
+	eDependencies := EventsDefinitions[e].Dependencies
+	for _, req := range eDependencies.requirements {
+		switch req {
+		case Ksymbols:
+			t.config.RequiredInits.kallsyms = true
```

Comment:
I think that instead of setting this config (which is actually intended for use by whoever uses the ebpf package, and not for internal use inside the package itself), you should aggregate these symbols to initialize the bpf map later.

Reply:
I think that the current situation is not as you describe.
If the events channel, error channel and the kconfigs are part of the configuration, it means that the use of the config is no longer to represent the configuration of the user, but to represent the state of tracee or to be its way of communication outside.
Anyway, I can do as you say. It is not so important to use config here (just making it more organized - a part is responsible for handling events info and another to load information like kallsyms). I do think that if this is how you think of the configuration maybe we should divide it to environment and configuration.